File: app/src/main/java/osama/me/viewmodelannotations/BaseViewHolder.java
package osama.me.viewmodelannotations;

import android.support.v7.widget.RecyclerView;
import android.view.View;


public abstract class BaseViewHolder<T> extends RecyclerView.ViewHolder {

    private T boundItem;

    public BaseViewHolder(final View itemView) {
        super(itemView);
    }

    protected void bind(T item) {
        boundItem = item;
    }

}

File: app/src/main/java/osama/me/viewmodelannotations/BaseViewHolder.kt
package osama.me.viewmodelannotations

import android.support.v7.widget.RecyclerView
import android.view.View


abstract class BaseViewHolder<T>( itemView: View) : RecyclerView.ViewHolder(itemView) {

    private var boundItem : T? = null

    open fun bind(item: T?) {
        boundItem = item
    }
}